client/src/components/common/UserMenu/index.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+import MenuItem from '@material-ui/core/MenuItem';
+import Menu from '@material-ui/core/Menu';
+
+const UserMenu = () => {
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const open = Boolean(anchorEl);
+
+  const handleMenu = (event) => {
```

Comment:
@alidah0 You can use `useCallback` to declare functions inside functional components that use hooks, and then your component will only declare that function once in memory and will reuse it, instead of redeclaring it every time your component renders.
see here: https://reactjs.org/docs/hooks-reference.html#usecallback

Otherwise, with the current code, `handleMenu` and `handleClose` are being re-created in memory every time this component re-renders.

Reply:
@tomduggan85  thank you but actually I got stuck at this, just new way to me 
I couldn't find what dependency should I pass ?? 
is this alright ?? 

```
  const handleMenu = useCallback((event) => {
    setAnchorEl(event.currentTarget);
  }, []);

  const handleClose = useCallback(() => {
    setAnchorEl(null);
  }, []);

```
instead of .. 

```
  const handleMenu = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

```

Reply:
Yes that looks correct to me.  Dependencies are optional, and many times (like this time) you won't need them.
If your callback used props or state, those props or state would go in the dependencies array.